ratelimit.go

```diff
@@ -63,24 +56,14 @@ func New(rate int, opts ...Option) Limiter {
 	l := &limiter{
 		perRequest: time.Second / time.Duration(rate),
 		maxSlack:   -10 * time.Second / time.Duration(rate),
+		timer:      time.NewTimer(time.Duration(math.MaxInt64)),
```

Comment:
rather than use a max duration, should we create a timer, stop it (maybe in `init`) and use that everywhere?

Reply:
I can also leave the `timer` field  as `nil` within the struct and have a check for it in `Take` & set it when needed - but I'm not sure if there are any performance implications.

Reply:
`nil` check works too, but less branches is always nice, so might be better to have an expired timer.